spacksites/spack-env-templates/slurm/gpu_first.yaml

```diff
@@ -0,0 +1,20 @@
+spack:
+
+  specs:
+  # CUDA packages, compile them with gcc@12
+  - matrix:
+    - - 'cuda@12.2.2' # v12.2.2 matches the latest CUDA installed on Young
```

Comment:
As mentioned on Slack, this is not necessarily the final version of `cuda` we'll use, I'm just mirroring the latest `cuda` module I found on current Young, but this is up for being changed.